### DISCUSSION TITLE
Facing trouble with the codegen

### DISCUSSION THREAD

**Discussion by programVeins** [maticzav/swift-graphql#67] [General]
I just wanted to point out a few problems I faced while trying to generate swift code for [OSlash](https://www.oslash.com)'s GraphQL schema. Tried running `swift-graphql ${ENDPOINT} --authorization ${AUTH}` but it resulted in the following error:

```swift
Error: valueNotFound(
    Swift.UnkeyedDecodingContainer,
    Swift.DecodingError.Context(
        codingPath: [
            CodingKeys(stringValue: "data", intValue: nil),
            CodingKeys(stringValue: "__schema", intValue: nil),
            CodingKeys(stringValue: "types", intValue: nil),
            _JSONKey(stringValue: "Index 10", intValue: 10),
            CodingKeys(stringValue: "interfaces", intValue: nil)
        ],
        debugDescription: "Cannot get unkeyed decoding container -- found null value instead.",
        underlyingError: nil
    )
)
```

I was wondering why this is being triggered because, swift-graphql uses `decoder.container(keyedBy: CodingKeys.self)` and the error says that an unkeyed decoding container was tried to parse. Does this mean the schema I am trying to parse has a few unkeyed nested containers? Will be cool if there's any known workaround for this! 🤔

**Comment by maticzav** [comment]
@programVeins any chance that you paste here the result of calling [the introspection query](https://github.com/maticzav/swift-graphql/blob/main/Sources/GraphQLAST/Introspection.swift) on your server? You can also send it to me privately.

I suspect the issue is that you have no interfaces and return `null` but the decoder expects an array.

**Reply by programVeins** [reply]
Hey, I realized #66 can fix this issue, I did, in fact, tweak a local copy of the source code and it now works as expected! You were right about interfaces, they seem to have caused the issue.

**Comment by shaps80** [comment]
Sounds like this was resolved and is no longer an issue, will close this issue to cleanup 👍
